AUTH-438: Added documentation for ZIP distribution (#115)
Alfresco/alfresco-identity-service: README.md

2. Choose the [sample realm](./alfresco-realm.json) file and click the "Create" button.

## Upgrading from Identity Service 1.1 to 1.2

  **_NOTE:_** The upgrade of the Alfresco Identity Management Service requires downtime. 
  This means that no user will be able to connect to any of the Digital Business Platform components while the upgrade or rollback is being done.

### General upgrade procedure

For upgrading Alfresco Identity Management Service we are mainly following the Keycloak upgrade procedure.

> We will be explaining how to do it if you are using our out of the box distribution or Kubernetes deployment.

Reviewer: > ... using our zip distribution or ...

> However depending on the environment you are using you should follow the next steps:

Reviewer: > you should follow these high-level steps:

> 1. Prior to applying the upgrade, handle any open transactions and delete the data/tx-object-store/ transaction directory.

Reviewer: Same [comment as on the other PR](https://github.com/Alfresco/alfresco-identity-service/pull/113#discussion_r353145678)

I suppose we'll have merge conflicts with this sections that we'll need to resolve depending on which PR gets in first.

2. Back up the old installation (configuration, themes, and so on).

3. Back up the database. For detailed information on how to back up the database, see the documentation for the relational database you are using.

4. Upgrade Keycloak server.

   - Testing the upgrade in a non-production environment first, to prevent any installation issues from being exposed in production, is a best practice.

   - Be aware that after the upgrade the database will no longer be compatible with the old server

   - Ensure the upgraded server is functional before upgrading adapters in production.

5. If you need to revert the upgrade, first restore the old installation, and then restore the database from the backup copy.

6. Upgrade the adapters.

Within the next sections we will go trough a simple distribution and kubernetes upgrade plus rollback.

  **_NOTE:_** In depth documentation on keycloak upgrade can be found [here](https://www.keycloak.org/docs/7.0/upgrading/index.html#_upgrading).

### ZIP Distribution

> #### Upgrade example for Identity Service with Postgres DB

Reviewer 2: ```suggestion
#### Upgrade example for Identity Service with PostgreSQL database
```

1. Backup the old installation performing:

	```bash

> 	pg_dump --clean --no-owner --no-acl -h ${POSTGRES_HOST} -p ${POSTGRES_PORT}  -U ${POSTGRES_USER} ${POSTGRES_DATABASE} | grep -v -E '(DROP\ SCHEMA\ public|CREATE\ SCHEMA\ public|COMMENT\ ON\ SCHEMA\ public|DROP\ EXTENSION\ plpgsql|CREATE\ EXTENSION\ IF\ NOT\ EXISTS\ plpgsql|COMMENT\ ON\ EXTENSION\ plpgsql)' > /backup/backup.sql

Reviewer: Doesn't our zip distribution use H2 by default?

Should we have a note that says the instructions assume a postrgresql has been used in a production environment?

Author: These instructions are under Upgrade example for Identity Service with PostgreSQL database. Is it ok? I think in the other merge request there's something written about the default H2 DB.

Reviewer: Ah, yes, I guess I missed that header due to the PR rendering.

	```
	

> 2. Remove old data and stop the postgres instance.

Reviewer 2: ```suggestion
2. Remove old data and stop the PostgreSQL instance.
```

3. Stop the Identity Service 1.1 server.

> 4. Open the 1.2 Identity Service distribution zip and configure accordingly to the database that will be used (for this example Postgres).

Reviewer 2: ```suggestion
4. Open Identity Service 1.2 distribution zip and configure accordingly to the database that will be used (for this example PostgreSQL).
```

   For detailed information on how to set up the desired database this visit the official documentation of Keycloak [here](https://www.keycloak.org/docs/4.8/server_installation/#_database).
   
5. Start the database and restore data by executing the following command:

 ```bash
  psql -h ${POSTGRES_HOST} -p ${POSTGRES_PORT} -d ${POSTGRES_DATABASE} -U ${POSTGRES_USER} -f /backup/backup.sql
 ``` 

> 6. Start Identity Service 1.2.

Reviewer 2: How do you start Identity Service?
